pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -291,23 +288,56 @@ func (pl *PodTopologySpread) isSchedulableAfterNodeChange(logger klog.Logger, po
 		return framework.Queue, err
 	}
 
+	// We return Queue when either of the following is met -
+	// 1. Node/Delete: the original node had a topology key that matches the pod's topology spread constraints.
+	// 2. Node/UpdateNodeLabels:
+	//  - the original node had a topology key that matches the pod's topology spread constraints,
+	//    but the modified node does not.
+	//  - the modified node has a topology key that matches the pod's topology spread constraints,
+	//    but the original node does not.
+	// 3. Node/Add: the created node has a topology key that matches the pod's topology spread constraints.
+
+	if modifiedNode == nil {
+		// framework.Delete: return Queue when node has topologyKey in its labels, else return QueueSkip.
+		if !nodeLabelsMatchSpreadConstraints(originalNode.Labels, constraints) {
+			logger.V(5).Info("the deleted node doesn't match pod topology spread constraints",
+				"pod", klog.KObj(pod), "node", klog.KObj(originalNode))
+			return framework.QueueSkip, nil
+		}
+		logger.V(5).Info("node that matches topology spread constraints was deleted, and the pod may be schedulable now",
+			"pod", klog.KObj(pod), "node", klog.KObj(originalNode))
+		return framework.Queue, nil
+	}
```

Comment:
Actually, can we move this to L339 to reduce the diff of this PR? Then we wouldn't need `if modifiedNode == nil` in the first place.

Reply:
done